homeassistant/components/template/coordinator.py

```diff
@@ -91,16 +99,41 @@ async def _attach_triggers(self, start_event=None) -> None:
             start_event is not None,
         )
 
-    async def _handle_triggered_with_script(self, run_variables, context=None):
+    async def _handle_triggered_with_script(
+        self, run_variables: TemplateVarsType, context: Context | None = None
+    ) -> None:
+        if not self._check_condition(run_variables):
+            return
         # Create a context referring to the trigger context.
         trigger_context_id = None if context is None else context.id
         script_context = Context(parent_id=trigger_context_id)
+        if TYPE_CHECKING:
+            # This method is only called if there's a script
+            assert self._script is not None
         if script_result := await self._script.async_run(run_variables, script_context):
             run_variables = script_result.variables
-        self._handle_triggered(run_variables, context)
+        self._execute_update(run_variables, context)
+
+    async def _handle_triggered(
+        self, run_variables: TemplateVarsType, context: Context | None = None
+    ) -> None:
+        if not self._check_condition(run_variables):
+            return
+        self._execute_update(run_variables, context)
+
+    def _check_condition(self, run_variables: TemplateVarsType) -> bool | None:
```

Comment:
This can't return `None` as far as I can tell.
```suggestion
    def _check_condition(self, run_variables: TemplateVarsType) -> bool:
```

Reply:
Good catch.